app/grandchallenge/evaluation/models.py

```diff
@@ -1173,12 +1173,20 @@ class Meta:
         unique_together = (("phase", "predictions_file", "algorithm_image"),)
 
     @cached_property
-    def is_evaluated_with_active_image(self):
+    def is_evaluated_with_active_image_and_ground_truth(self):
         active_image = self.phase.active_image
+        active_ground_truth = self.phase.active_ground_truth
         if active_image:
-            return Evaluation.objects.filter(
-                submission=self, method=active_image
-            ).exists()
+            if active_ground_truth:
+                return Evaluation.objects.filter(
+                    submission=self,
+                    method=active_image,
+                    ground_truth=active_ground_truth,
+                ).exists()
+            else:
+                return Evaluation.objects.filter(
+                    submission=self, method=active_image
+                ).exists()
```

Comment:
Doesn't this work?

```suggestion
            return Evaluation.objects.filter(
                submission=self,
                method=active_image,
                ground_truth=active_ground_truth,
            ).exists()
```

Reply:
Ah, yes, that works and is necessary, or else we match too many evaluations